upload/common/upload_area.py

```diff
@@ -263,6 +269,24 @@ def retrieve_file_count_for_upload_area(self):
         results = query_result.fetchall()
         return results[0][0]
 
+    def _get_file_if_exists(self, filename, clientside_checksums):
```

Comment:
I think this should live in `UploadedFile`, not here.
It would be nice if you could say `file = UploadedFile(filename)` and then `file.exists` but UploadedFile doesn't work that way - it must always exist.  However I still think this should perhaps be a class method on UploadedFile. As we have a class for files, then code that checks if a file exists should live there.

Reply:
Done.